posthog/queries/sessions.py

```diff
@@ -0,0 +1,271 @@
+
+            hash_ids = []
+            for session in sessions:
+                for event in session["events"]:
+                    if event.get("elements_hash"):
+                        hash_ids.append(event["elements_hash"])
+
+            groups = self._prefetch_elements(hash_ids, team)
+
+            for session in sessions:
+                for event in session["events"]:
+                    try:
+                        event.update(
+                            {
+                                "elements": ElementSerializer(
+                                    [group for group in groups if group.hash == event["elements_hash"]][0]
+                                    .element_set.all()
+                                    .order_by("order"),
+                                    many=True,
+                                ).data
+                            }
+                        )
+                    except IndexError:
+                        event.update({"elements": []})
+        return sessions
+
+    def _session_avg(self, base_query: str, params: Tuple[Any, ...], filter: Filter) -> List[Dict[str, Any]]:
+        def _determineInterval(interval):
+            if interval == "minute":
+                return (
+                    "minute",
+                    "min",
+                )
+            elif interval == "hour":
+                return "hour", "H"
+            elif interval == "week":
+                return "week", "W"
+            elif interval == "month":
+                return "month", "M"
+            else:
+                return "day", "D"
+
+        interval, interval_freq = _determineInterval(filter.interval)
+
+        average_length_time = "SELECT date_trunc('{interval}', timestamp) as start_time,\
+                        AVG(length) AS average_session_length_per_day,\
+                        SUM(length) AS total_session_length_per_day, \
+                        COUNT(1) as num_sessions_per_day\
+                        FROM (SELECT global_session_id, EXTRACT('EPOCH' FROM (MAX(timestamp) - MIN(timestamp)))\
+                            AS length,\
+                            MIN(timestamp) as timestamp FROM ({}) as count GROUP BY 1) as agg group by 1 order by start_time".format(
+            base_query, interval=interval
+        )
+
+        cursor = connection.cursor()
+        cursor.execute(average_length_time, params)
+        time_series_avg = cursor.fetchall()
+
+        date_range = pd.date_range(filter.date_from, filter.date_to, freq=interval_freq,)
+        df = pd.DataFrame([{"date": a[0], "count": a[1], "breakdown": "Total"} for a in time_series_avg])
+        if interval == "week":
+            df["date"] = df["date"].apply(lambda x: x - pd.offsets.Week(weekday=6))
+        elif interval == "month":
+            df["date"] = df["date"].apply(lambda x: x - pd.offsets.MonthEnd(n=0))
+
+        df_dates = pd.DataFrame(df.groupby("date").mean(), index=date_range)
```

Comment:
compare isn't working when it reaches this line. This also appears to have been broken before prior to this PR

Reply:
Fixed, was actually an issue with empty Event querysets in general